ocaml/testsuite/tests/typing-higher-jkinds/top.ml

```diff
@@ -0,0 +1,66 @@
+(* TEST
+ flags = "-extension layouts_alpha";
+ expect;
+*)
+
+type t : top
+[%%expect {|
+type t : top
+|}]
+
+type t : top => value
+[%%expect {|
+type t : top => value
+|}]
+
+module M : sig
+  type t : top
+end = struct
+  type t : top
+end
+[%%expect {|
+module M : sig type t : top end
+|}]
+
+module M : sig
+  type t : top
+end = struct
+  type t : value
+end
+[%%expect {|
+module M : sig type t : top end
+|}]
```

Comment:
It would be worth adding a test that shows that arrow kinds can also subkind to `top`.